docs/usage-notes.md

```diff
@@ -0,0 +1,52 @@
+This document shows several usage scenarios of SMACK that require special flags.
```

Comment:
I wonder if you should also add a paragraph or two explaining unrolling (with an example).